Automattic/wp-calypso
# Analytics: do not omit site id when a site fragment is present in the path

This PR builds on top of #23728 to assume that when analytics paths are reported **correctly**, we only include the blog id as part of the payload when a site fragment `:site` is available. Eg `/comments/all/:site` but not for `/comments`.

On further investigation when we call `shouldReportOmitBlogId` the input given is the normalized path rather than the original one. Since this is the case we can also remove the section blacklist.

### Testing Instructions
- Enable debugging `localStorage.setItem('debug', 'calypso:analytics:tracks')`
- When visiting urls with a site fragment we see a `blog_id` that matches the slug
- When visiting urls without a site fragment we `blog_id` is not set.
- Click on My Sites and select a site. **Without refreshing/or pasting in the url** click on your profile, or visit the reader. We should not see blog_id set.

<img width="1287" alt="screen shot 2018-03-30 at 2 17 39 pm" src="https://user-images.githubusercontent.com/1270189/38154309-36021262-3426-11e8-9f89-afd186f0b0d5.png">

Needed as part of #23647 

cc @yanirs if you prefer this general behavior or if you would prefer a blacklist for particular sections. For the latter there are corner cases with paths like `/me/purchases/:site/:purchaseId/cancel-privacy-protection` because it is in the profile section.
